acceptance/suites/tests/authorization/x509_auth.rb

```diff
@@ -0,0 +1,139 @@
+test_name "(TK-293) TK-AUTH uses certificate extensions for authentication" do
+
+confine :except, :platform => 'windows'
+  
+server = master.puppet['certname']
+ssldir = master.puppet['ssldir']
+confdir = master.puppet['confdir']
+
+teardown do
+  # restore the original tk auth.conf file
+  on master, 'cp /etc/puppetlabs/puppetserver/conf.d/auth.bak /etc/puppetlabs/puppetserver/conf.d/auth.conf'
+end
+
+step "Backup the tk auth.conf file" do
+  on master, 'cp /etc/puppetlabs/puppetserver/conf.d/auth.conf /etc/puppetlabs/puppetserver/conf.d/auth.bak'
+end
+
+# Do we have a functioning cert?
+step "Confirm agent can connect with existing cert" do
+  agents.each do |a|
+    if (not_controller(a))
+      rc = on(a,
+              puppet("agent --test --server #{server} --detailed-exitcodes"),
+              {:acceptable_exit_codes => [0,2]})
+    end
+  end
+end
+
+# Not anymore we don't
+step "Revoke and destroy the existing cert on the server" do
+  agents.each do |a| 
+    if (not_controller(a))
+      nodename = fact_on(a, 'fqdn')
```

Comment:
It is true that `hostname` and `certname` are the same *most* of the time, but this is not guaranteed. This is an edge case and I don't know if it is worth guarding against, but I wanted to bring it up.

Example
```
  sol10.delivery.puppetlabs.net (solaris-10-sparc) 13:16:14$ puppet agent --test --server v2u8phod7ts8nak.delivery.puppetlabs.net
    Info: Creating a new SSL key for unknown.delivery.puppetlabs.net
    Info: Caching certificate for ca
    Info: csr_attributes file loading from /etc/puppetlabs/puppet/csr_attributes.yaml
    Info: Creating a new SSL certificate request for unknown.delivery.puppetlabs.net
    Info: Certificate Request fingerprint (SHA256): BE:62:68:35:61:75:A7:49:1C:DE:94:98:73:5D:34:F8:FE:19:D8:04:ED:8D:B4:4E:33:F3:4E:7D:B6:FF:E9:31
    Info: Caching certificate for ca
    Exiting; no certificate found and waitforcert is disabled
```